src/rospkg/environment.py

```diff
@@ -109,7 +109,7 @@ def get_ros_package_path(env=None):
     """
     if env is None:
         env = os.environ
-    return env.get(ROS_PACKAGE_PATH, None)
+    return env.get(ROS_PACKAGE_PATH, '') + ':/usr/share'
```

Comment:
* William Woodall <notifications@github.com> [2016-08-10 15:35]:
> Sure, I think I can demonstrate how it can be logically inconsistent, and a come up with a hypothetical scenario where that inconsistency could actually actually break something.
> 
> If you have packages A, B, and C installed from Ubuntu (into `/usr`) and you have A, B, and D installed in `/opt/ros/kinetic`, then you do `rospack list` you'll get A, B, and C. However, if you source the `/opt/ros/kinetic/setup.bash` and do `rospack list` you'll get A, B, C, and D. Where C comes from `/usr` and the others point to `/opt/ros/kinetic`. Let's say that both C and D build against (use headers from and link against) B.
> 
> This can be a problem if, for example, you build a package locally called E which uses both C and D.
> 
> First consider you do this in a clean shell. The build should fail because D is not found.
> 
> Then consider you try to build it after sourcing the `/opt/ros/kinetic/setup.bash` file. Then the build does not fail because it finds C in `/usr` and D in `/opt/ros/kinetic`. The "expected" behavior would be that the build fails because C is not found, but that's not what happens. However, even if the build works (might not due to differences in headers between versions of B), it very likely will have problems when running since C is built against B in `/usr` and D is built against B in `/opt/ros/kinetic` (let's call that B'). If B and B' are not ABI compatible, then you've got a problem.
> 
> This same issue can be extended to Python code, cmake modules, locating resources from other packages, etc.

That's not a ROS problem (or one with this patch after all). Just
replace "package" with "plain C library" and you get the same problem.
Generally it is a problem if users try to mix different package sources.

I don't think we can find a solution here, as long as people are able to
install conflicting versions at the same time (I've worked on a more
general solution to this already, but I don't think this is the right
place to discuss it).

> This happened because the version of `rospkg` that got installed didn't have this patch and so it couldn't find `roslaunch` in `/usr`. That's why I was so interested in finding a way to satisfy the Debian use case with a patch that we could merge upstream -- to avoid this issue.

Yes, so we agree that it would be great to merge this patch upstream and
don't run into it again :).